flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSplitGenerator.java

```diff
@@ -38,7 +48,22 @@ private FlinkSplitGenerator() {
     List<CombinedScanTask> tasks = tasks(table, context);
     FlinkInputSplit[] splits = new FlinkInputSplit[tasks.size()];
     for (int i = 0; i < tasks.size(); i++) {
-      splits[i] = new FlinkInputSplit(i, tasks.get(i));
+      Set<String> hosts = Sets.newHashSet();
+      CombinedScanTask combinedScanTask = tasks.get(i);
+      combinedScanTask.files().forEach(fileScanTask -> {
+        try {
+          final FileSystem fs = new HadoopFileSystem(DistributedFileSystem.get(new Configuration()));
```

Comment:
This should do the same thing that Spark does. Specifically:
* There should be a flag to enable locality
* The flag should be defaulted based on whether the scheme identifies a FS that has locality
* `new Configuration()` should not be called
* Use `Util.blockLocations` to get block locations
* Ideally, parallelize the block lookup per split because it can get expensive

Reply:
- I added this flag to the Build of FlinkSource  and determined the value of this flag based on the FS and configuration. But I'm quit not sure it's appropriate to add a flag there.
- `Util.blockLocations`  is indeed a better choice.

PTAL, thx.